front/src/applications/editor/tools/pointEdition/utils.ts

```diff
@@ -78,23 +80,41 @@ function getSignalDefaultSettings(signalingSystem?: 'BAL' | 'BAPR' | 'TVM' | und
 export function formatSignalingSystems(
   entity: SignalEntity
 ): SignalEntity['properties']['logical_signals'] {
-  return entity.properties.logical_signals
-    ? entity.properties.logical_signals.map((logicalSignal) => {
-        const next_signaling_systems =
-          logicalSignal && logicalSignal.next_signaling_systems
-            ? logicalSignal.next_signaling_systems.map(
-                (nextSignalingSystem) => nextSignalingSystem || 'BAL'
-              )
-            : [];
-        const settings = {
-          ...getSignalDefaultSettings(logicalSignal?.signaling_system),
-          ...(logicalSignal?.settings || {}),
-        };
-        return {
-          signaling_system: logicalSignal?.signaling_system || 'BAL',
-          next_signaling_systems,
-          settings,
-        } as SignalingSystem;
-      })
-    : [];
+  if (!entity.properties.logical_signals) return [];
+
+  return entity.properties.logical_signals.map((logicalSignal) => {
+    // console.log('logicalSignal:', logicalSignal);
+    const next_signaling_systems =
+      logicalSignal && logicalSignal.next_signaling_systems
+        ? logicalSignal.next_signaling_systems.map(
+            (nextSignalingSystem) => nextSignalingSystem || 'BAL'
+          )
+        : [];
+    const settings = {
+      ...getSignalDefaultSettings(logicalSignal?.signaling_system),
+      ...(logicalSignal?.settings || {}),
+    };
```

Comment:
If you have changed the type, then you can simplify the code (remove the ?, the [] and the {})
```suggestion
    const settings = {
      ...getSignalDefaultSettings(logicalSignal.signaling_system),
      ...logicalSignal.settings,
    };
```